src/sage/categories/finite_groups.py

```diff
@@ -101,12 +101,16 @@ def cardinality(self):
                 sage: G.cardinality()
                 384
             """
+
+            # Note for developers - self will have _deg attribute if it is derived from  MatrixGroup_generic class
+            if hasattr(self, '_deg'):
+                return self._deg
```

Comment:
Why do you return the degree of the group for its cardinality?

Reply:
Thank you for pointing that out! I just saw that it is a parent method for all finite groups, so doing this won't work. But I propose implementing a cardinality function specifically for the special linear group, that will fix all the issues.

Reply:
So for now, I am converting this PR to a draft until I implement the method.